### PR TITLE
cleanup headers

### PR DESCRIPTION
**Pull request by lukileczo**
## Description

Cleanup header files:
- move definitions shared between user- and kernel-space to kernel headers and install appropriate files, including constant-width integer types
- cleanup arch-dependent includes
- update types in syscalls which use `handle_t`
- remove `offs_t` type
- install kernel headers on `host` target

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing - verifying PR https://github.com/phoenix-rtos/phoenix-rtos-project/pull/1001
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work https://github.com/phoenix-rtos/libphoenix/pull/339
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: include/posix-stat.h [hunk 1]
```diff
@@ -136,81 +91,4 @@ struct stat {
...
-
-
-struct cmsghdr {
-	socklen_t cmsg_len;
-	int cmsg_level;
-	int cmsg_type;
-};
-
-#endif
+#endif
```
Comment (agkaminski) [inline include/posix-stat.h:94]: Missing new line

File: proc/userintr.c [hunk 2]
```diff
@@ -90,7 +90,7 @@ static int userintr_dispatch(unsigned int n, cpu_context_t *ctx, void *arg)
 }
 
 
-int userintr_setHandler(unsigned int n, int (*f)(unsigned int, void *), void *arg, unsigned int c)
+int userintr_setHandler(unsigned int n, int (*f)(unsigned int, void *), void *arg, int c)
```
Comment (agkaminski) [inline proc/userintr.c:93]:
```suggestion
int userintr_setHandler(unsigned int n, int (*f)(unsigned int, void *), void *arg, handle_t c)
```

File: proc/userintr.h [hunk 3]
```diff
@@ -34,7 +34,7 @@ typedef struct _userintr_t {
 extern void userintr_put(userintr_t *ui);
 
 
-extern int userintr_setHandler(unsigned int n, int (*f)(unsigned int, void *), void *arg, unsigned int c);
+extern int userintr_setHandler(unsigned int n, int (*f)(unsigned int, void *), void *arg, int c);
```
Comment (agkaminski) [inline proc/userintr.h:37]:
```suggestion
extern int userintr_setHandler(unsigned int n, int (*f)(unsigned int, void *), void *arg, handle_t c);
```

File: syscalls.c [hunk 4]
```diff
@@ -264,7 +263,7 @@ int syscalls_beginthreadex(void *ustack)
 	void (*start)(void *);
 	unsigned int priority, stacksz;
 	void *stack, *arg;
-	unsigned int *id;
+	handle_t *id;
```
Comment (agkaminski) [inline syscalls.c:266]: tid is not a handle_t, should be plain int
```suggestion
	int *id;
```

File: syscalls.c [hunk 5]
```diff
@@ -273,7 +272,7 @@ int syscalls_beginthreadex(void *ustack)
 	GETFROMSTACK(ustack, void *, stack, 2);
 	GETFROMSTACK(ustack, unsigned int, stacksz, 3);
 	GETFROMSTACK(ustack, void *, arg, 4);
-	GETFROMSTACK(ustack, unsigned int *, id, 5);
+	GETFROMSTACK(ustack, handle_t *, id, 5);
```
Comment (agkaminski) [inline syscalls.c:275]:
```suggestion
	GETFROMSTACK(ustack, int *, id, 5);
```

**Review by agkaminski — DISMISSED** [review]

**Review by anglov — APPROVED** [review]
LGTM

File: hal/ia32/arch/types.h [hunk 6]
```diff
@@ -17,52 +17,22 @@
...
-typedef offs_t off_t;
+typedef __u8 ld80[10]; /* Long double as bytes */
 
-typedef unsigned int size_t;
-typedef unsigned long long time_t;
+typedef __u64 cycles_t;
+typedef __u32 ptr_t;
 
-typedef u32 ptr_t;
+typedef unsigned long size_t;
```
Comment (gerard5) [inline hal/ia32/arch/types.h:32]: `ssize_t` for IA32 is defined to be `int` (not long) in current PR ([include/arch/ia32/types.h:29](https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/518/files#diff-3036bc66d98d2f7860b3b162b835ad5c753b76ec76be4963c9a67dd053865151R29))
Why `size_t` typedef changed to `unsigned long int` from`unsigned int` on IA32  ?

Comment (lukileczo) [reply hal/ia32/arch/types.h:32]: I checked how the toolchain defines `size_t` on IA32 and in `stddef.h` which is used in userspace it's defines as `unsigned long` so I've changed it, thanks for spotting the `ssize_t`

**Review by agkaminski — APPROVED** [review]

**Pull request merged**